### PR TITLE
RHCLOUD-33780 | feature: Kessel tests for the "EndpointResource"

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
Implements the tests and the required helper classes to test making
authenticated and authorized requests while using Kessel as the
authentication and authorization back end.

## Dependency
- https://github.com/RedHatInsights/notifications-backend/pull/3032

## Jira ticket
[[RHCLOUD-33780]](https://issues.redhat.com/browse/RHCLOUD-33780)

### CODE REVIEW

**Comment by MikelAlejoBR** [comment]
I'm going to separate the changes because this is unreviewable.

**Review by g-duval — COMMENTED** [review]
Thanks a lot for this contribution @MikelAlejoBR !
I know how painful is applying the same thing on dosen of methods.

File: backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java
Comment (MikelAlejoBR) [reply backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java:2577]: I agree, yes. I have modified the code so that we return `404`s when the endpoints do not exist... the problem I see is that many functions fetch the endpoint at a later stage, so we end up doing kind of the following:

```java
public void handler(@Param final UUID endpointId) {
    if (!endpointRepository.existsById(endpointId)) {
        throw new NotFoundException();
    }

     // ... If Kessel this, if RBAC that and then...
    innerFunction();
}

public void innerFunction(final UUID endpointId) {
    // ...
    final Endpoint dbEndpoint = endpointRepository.getById(endpointId);
    if (dbEndpoint == null) {
        throw new NotFoundException();
    }
    // ...
}
```

So we have two database calls for basically the same thing. So I wonder if I should extract all those database calls and move them to the initial handlers instead, and then pass the fetched endpoint directly to the inner functions :thinking: ... But maybe that should be part of a different PR, what do you think?

Comment (g-duval) [reply backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java:2584]: Ok, I think we can keep it simple as it is then.

**Review by g-duval — APPROVED** [review]

**Comment by MikelAlejoBR** [comment]
Once the tests pass I'm going to merge it to unblock the other PRs, but I'll need to review the tests and the permissions one more time for the `EndpointResource` :cry: :sob:

**Pull request merged**
